espnet2/gan_svs/post_frontend/fused.py

```diff
@@ -0,0 +1,155 @@
+
+        for i, postfrontend in enumerate(postfrontends):
+            postfrontend_type = postfrontend["postfrontend_type"]
+            if postfrontend_type == "default":
+                n_mels, fs, n_fft, win_length, hop_length = (
+                    postfrontend.get("n_mels", 80),
+                    fs,
+                    postfrontend.get("n_fft", 512),
+                    postfrontend.get("win_length"),
+                    postfrontend.get("hop_length", 128),
+                )
+                window, center, normalized, onesided = (
+                    postfrontend.get("window", "hann"),
+                    postfrontend.get("center", True),
+                    postfrontend.get("normalized", False),
+                    postfrontend.get("onesided", True),
+                )
+                fmin, fmax, htk, apply_stft = (
+                    postfrontend.get("fmin", None),
+                    postfrontend.get("fmax", None),
+                    postfrontend.get("htk", False),
+                    postfrontend.get("apply_stft", True),
+                )
+
+                self.postfrontends.append(
+                    DefaultFrontend(
+                        n_mels=n_mels,
+                        n_fft=n_fft,
+                        fs=fs,
+                        win_length=win_length,
+                        hop_length=hop_length,
+                        window=window,
+                        center=center,
+                        normalized=normalized,
+                        onesided=onesided,
+                        fmin=fmin,
+                        fmax=fmax,
+                        htk=htk,
+                        apply_stft=apply_stft,
+                    )
+                )
+            elif postfrontend_type == "s3prl":
+                postfrontend_conf, download_dir, multilayer_feature = (
+                    postfrontend.get("postfrontend_conf"),
+                    postfrontend.get("download_dir"),
+                    postfrontend.get("multilayer_feature"),
+                )
+                self.postfrontends.append(
+                    S3prlPostFrontend(
+                        fs=fs,
+                        input_fs=input_fs,
+                        postfrontend_conf=postfrontend_conf,
+                        download_dir=download_dir,
+                        multilayer_feature=multilayer_feature,
+                    )
+                )
+
+            else:
+                raise NotImplementedError  # frontends are only default or s3prl
+
+        self.postfrontends = torch.nn.ModuleList(self.postfrontends)
+
+        self.gcd = np.gcd.reduce(
+            [postfrontend.hop_length for postfrontend in self.postfrontends]
+        )
+        self.factors = [
+            postfrontend.hop_length // self.gcd for postfrontend in self.postfrontends
+        ]
+        if torch.cuda.is_available():
+            dev = "cuda"
+        else:
+            dev = "cpu"
```

Comment:
Since it is a sub-module of the gan_svs model, we may not need to manually add the device setup?